File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        unordered_map<int,int>prefix;
        vector<int> output(nums.size());

        int m = 1;
        for(int i = 0;i<nums.size();i++){
            prefix[i] = m;
            m = m*nums[i];
        }
        m = 1;
        for(int j = nums.size()-1;j>=0;j--){
            output[j] = prefix[j]*m;
            m = m * nums[j];
        }
        return output;
    }
};

File: Data Structures & Algorithms/jump-game/submission-0.java
class Solution {
    public boolean canJump(int[] nums) {
        int i=0,h=0;
        while(i<nums.length-1){
            if(nums[i]>=h){
                h=nums[i];
            }
            if(h==0){
                return false;
            }
            h--;
            i++;
        }
        return true;
    }
}
